framework/guid/model.py

```diff
@@ -20,6 +20,26 @@ class Guid(StoredObject):
     _id = fields.StringField(primary=True)
     referent = fields.AbstractForeignField()
 
+    @classmethod
+    def generate(self, referent=None):
+        while True:
+            # Create GUID
+            guid_id = ''.join(random.sample(ALPHABET, 5))
```

Comment:
Have we considered using `random.SysRandom` instead of `random`?

While our GUIDs aren't being used for security purposes, using the Python implementation can generate overlapping sequences of values in some threaded contexts.

From [the Python docs](https://docs.python.org/2/library/random.html):

> The functions supplied by this module are actually bound methods of a hidden instance of the random.Random class. You can instantiate your own instances of Random to get generators that don’t share state. This is especially useful for multi-threaded programs, creating a different instance of Random for each thread, and using the jumpahead() method to make it likely that the generated sequences seen by each thread don’t overlap.

I don't think this is something that needs changed right now, but it could impact performance when attempting to generate GUIDs concurrently.